test/spec/modules/rubiconBidAdapter_spec.js

```diff
@@ -3286,6 +3338,91 @@ describe('the rubicon adapter', function () {
           expect(bids[0].cpm).to.be.equal(0);
         });
 
+        it('should handle DSA object from response', function() {
+          let response = {
+            'status': 'ok',
+            'account_id': 14062,
+            'site_id': 70608,
+            'zone_id': 530022,
+            'size_id': 15,
+            'alt_size_ids': [
+              43
+            ],
+            'tracking': '',
+            'inventory': {},
+            'ads': [
+              {
+                'status': 'ok',
+                'impression_id': '153dc240-8229-4604-b8f5-256933b9374c',
+                'size_id': '15',
+                'ad_id': '6',
+                'adomain': ['test.com'],
+                'advertiser': 7,
+                'network': 8,
+                'creative_id': 'crid-9',
+                'type': 'script',
+                'script': 'alert(\'foo\')',
+                'campaign_id': 10,
+                'cpm': 0.811,
+                'targeting': [
+                  {
+                    'key': 'rpfl_14062',
+                    'values': [
+                      '15_tier_all_test'
+                    ]
+                  }
+                ],
+                'dsa': {
+                  'behalf': 'Advertiser',
+                  'paid': 'Advertiser',
+                  'transparency': [{
+                    'domain': 'dsp1domain.com',
+                    'dsaparams': [1, 2]
+                  }],
+                  'adrender': 1
+                }
+              },
+              {
+                'status': 'ok',
+                'impression_id': '153dc240-8229-4604-b8f5-256933b9374d',
+                'size_id': '43',
+                'ad_id': '7',
+                'adomain': ['test.com'],
+                'advertiser': 7,
+                'network': 8,
+                'creative_id': 'crid-9',
+                'type': 'script',
+                'script': 'alert(\'foo\')',
+                'campaign_id': 10,
+                'cpm': 0.911,
+                'targeting': [
+                  {
+                    'key': 'rpfl_14062',
+                    'values': [
+                      '43_tier_all_test'
+                    ]
+                  }
+                ],
+                'dsa': {}
+              }
+            ]
+          };
+          // config.setConfig({
+          //   rubicon: {
+          //     netRevenue: false
+          //   }
+          // });
+          let bids = spec.interpretResponse({body: response}, {
+            bidRequest: bidderRequest.bids[0]
+          });
+          expect(bids).to.be.lengthOf(2);
+          expect(bids[1].meta.dsa).to.have.property('behalf');
+          expect(bids[1].meta.dsa).to.have.property('paid');
+
+          // if we dont have dsa field in response or the dsa object is empty
+          expect(bids[0].meta).to.not.have.property('dsa');
```

Comment:
seems strange that the input ads[0] has DSA and ads[1] does not

But bids[0] does not have DSA but ads[1] does.

But I guess it has something to do with the input `bidRequest` into the `spec.interpretResponse` above?

Anyways, seems like it works and tests pass

Reply:
The comments have been updated.